starter-code/memory.js

```diff
@@ -3,34 +3,81 @@
-  		{ name: "spiderman",       img: "spiderman.jpg" },
-  		{ name: "superman",        img: "superman.jpg" },
-  		{ name: "the avengers",    img: "the-avengers.jpg" },
-  		{ name: "thor",            img: "thor.jpg" },
-      { name: "aquaman",         img: "aquaman.jpg" },
-  		{ name: "batman",          img: "batman.jpg" },
-  		{ name: "captain america", img: "captain-america.jpg" },
-      { name: "fantastic four",  img: "fantastic-four.jpg" },
-  		{ name: "flash",           img: "flash.jpg" },
-  		{ name: "green arrow",     img: "green-arrow.jpg" },
-  		{ name: "green lantern",   img: "green-lantern.jpg" },
-  		{ name: "ironman",         img: "ironman.jpg" },
-  		{ name: "spiderman",       img: "spiderman.jpg" },
-  		{ name: "superman",        img: "superman.jpg" },
-  		{ name: "the avengers",    img: "the-avengers.jpg" },
-  		{ name: "thor",            img: "thor.jpg" },
-  	];
-    this.selectedCards = [];
-    this.pairsClicked = 0;
-    this.correctPairs = 0;
+    { name: "aquaman", img: "aquaman.jpg" },
+    { name: "batman", img: "batman.jpg" },
+    { name: "captain america", img: "captain-america.jpg" },
+    { name: "fantastic four", img: "fantastic-four.jpg" },
+    { name: "flash", img: "flash.jpg" },
+    { name: "green arrow", img: "green-arrow.jpg" },
+    { name: "green lantern", img: "green-lantern.jpg" },
+    { name: "ironman", img: "ironman.jpg" },
+    { name: "spiderman", img: "spiderman.jpg" },
+    { name: "superman", img: "superman.jpg" },
+    { name: "the avengers", img: "the-avengers.jpg" },
+    { name: "thor", img: "thor.jpg" },
+    { name: "aquaman", img: "aquaman.jpg" },
+    { name: "batman", img: "batman.jpg" },
+    { name: "captain america", img: "captain-america.jpg" },
+    { name: "fantastic four", img: "fantastic-four.jpg" },
+    { name: "flash", img: "flash.jpg" },
+    { name: "green arrow", img: "green-arrow.jpg" },
+    { name: "green lantern", img: "green-lantern.jpg" },
+    { name: "ironman", img: "ironman.jpg" },
+    { name: "spiderman", img: "spiderman.jpg" },
+    { name: "superman", img: "superman.jpg" },
+    { name: "the avengers", img: "the-avengers.jpg" },
+    { name: "thor", img: "thor.jpg" }
+  ];
+  this.selectedCards = [];
+  this.pairsClicked = 0;
+  this.correctPairs = 0;
+  this._shuffleCards();
+};
+
+//Shuffling
+MemoryGame.prototype._shuffleCards = function() {
+  this.cards = _.shuffle(this.cards);
+};
+
+//Selecting a Card
+MemoryGame.prototype.selectCard = function(card) {
+  console.log($(card).attr("name"));
+  if (this.selectedCards.length === 1) {
+    this.pairsClicked++;
+    $("#pairs_clicked").text(this.pairsClicked);
+    // update the dom
+    if ($(card).attr("name") === $(this.selectedCards[0]).attr("name")) {
+      this.selectedCards = [];
+      this.correctPairs++;
+      $("#pairs_guessed").text(this.correctPairs);
+      var messageY = function() {
+        alert("Good job man!");
+        $(".front").hide();
```

Comment:
This might be causing trouble. Notice how after a good guess, the cards don't stay flipped to the front, but instead hide and "mess up" the board.
Try defining methods for these cases and it should be easier to debug :)